dms/models/dms_security_mixin.py

```diff
@@ -133,6 +134,23 @@ def _apply_access_groups(self, query, mode="read"):
             where_clause = "({groups_clause} OR {exists_clause})".format(
                 groups_clause=where_clause, exists_clause=exists_clause,
             )
+        # Add _get_directory_ids_with_res_model_without_access
+        if not self.env.context.get("bypass_ids_with_res_model_without_access"):
+            custom_ids = self._get_directory_ids_with_res_model_without_access(mode)
+            if len(custom_ids) > 0:
+                field = "id"
+                if self._name == "dms.file":
+                    field = "directory_id"
+                extra_clause = """
+                    "{table}".{field} NOT IN {ids}
+                """.format(
+                    table=self._table,
+                    field=field,
+                    ids=", ".join(map(lambda id: "(%s)" % id, custom_ids)),
+                )
+                where_clause = "({where_clause} AND {extra_clause})".format(
+                    where_clause=where_clause, extra_clause=extra_clause,
+                )
```

Comment:
And by adding this line, we tell `psycopg2` to use `custom_ids` to expand those `%s` above.

```suggestion
                )
                query.where_clause_params += custom_ids
```